src/jit/emit.cpp

```diff
@@ -3048,11 +3056,12 @@ emitter::instrDesc *emitter::emitNewInstrCallDir(int        argCnt,
      */
     bool gcRefRegsInScratch = ((gcrefRegs & RBM_CALLEE_TRASH) != 0);
 
-    if  (!VarSetOps::IsEmpty(emitComp, GCvars)     ||   // any frame GCvars live
-         gcRefRegsInScratch          ||   // any register gc refs live in scratch regs
-         (byrefRegs != 0)            ||   // any register byrefs live
-         (argCnt > ID_MAX_SMALL_CNS) ||   // too many args
-         (argCnt < 0)                   ) // caller pops arguments
+    if  (!VarSetOps::IsEmpty(emitComp, GCvars)  ||                              // any frame GCvars live
+         gcRefRegsInScratch                     ||                              // any register gc refs live in scratch regs
+         (byrefRegs != 0)                       ||                              // any register byrefs live
+         (argCnt > ID_MAX_SMALL_CNS)            ||                              // too many args
+         (argCnt < 0)                                                           // caller pops arguments
+         FEATURE_UNIX_AMD64_STRUCT_PASSING_ONLY(|| (retSize1In != EA_UNKNOWN))) // There is a second ref/byref return register.
```

Comment:
I think that the correct check is:

EA_IS_GCREF(retSize1In) 

As we only need the larger call descriptor when we actually have a second return register that is a GCref.